### PR TITLE
Fix deprecated usage of lock_memory API

### PR DESCRIPTION
**Pull request by imsudiproy**
Contributions via pull requests are much appreciated. Before sending us a pull request, please ensure that:

1. Limited scope. Your PR should do one thing or one set of things. Avoid adding “random fixes” to PRs. Put those on separate PRs.
2. Give your PR a descriptive title. Add a short summary, if required.
3. Make sure the pipeline is green.
4. Don’t be afraid to request reviews from maintainers.
5. New code = new tests. If you are adding new functionality, always make sure to add some tests exercising the code and serving as live documentation of your original intention.

To send us a pull request, please:

- [x] Fork the repository.
- [x] Modify the source; please focus on the specific change you are contributing. If you also reformat all the code, it will be hard for us to focus on your change.
- [x] Ensure local tests pass. (`colcon test` and `pre-commit run` (requires you to install pre-commit by `pip3 install pre-commit`)
- [x] Commit to your fork using clear commit messages.
- [x] Send a pull request, answering any default questions in the pull request interface.
- [x] Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.

### CODE REVIEW

File: controller_manager/src/ros2_control_node.cpp [hunk 1]
```diff
@@ -61,10 +61,10 @@ int main(int argc, char ** argv)
 
   const bool has_realtime = realtime_tools::has_realtime_kernel();
   const bool lock_memory = cm->get_parameter_or<bool>("lock_memory", has_realtime);
-  std::string message;
-  if (lock_memory && !realtime_tools::lock_memory(message))
+  auto lock_result = realtime_tools::lock_memory();
```
Comment (christophfroehlich) [inline controller_manager/src/ros2_control_node.cpp:64]: this has to be called only if lock_memory parameter is true

**Review by christophfroehlich — CHANGES_REQUESTED** [review]
Thanks, but there is a logical issue here

File: controller_manager/src/ros2_control_node.cpp [hunk 2]
```diff
@@ -61,10 +61,12 @@ int main(int argc, char ** argv)
 
   const bool has_realtime = realtime_tools::has_realtime_kernel();
   const bool lock_memory = cm->get_parameter_or<bool>("lock_memory", has_realtime);
-  std::string message;
-  if (lock_memory && !realtime_tools::lock_memory(message))
+  if (lock_memory) 
   {
-    RCLCPP_WARN(cm->get_logger(), "Unable to lock the memory : '%s'", message.c_str());
+    auto lock_result = realtime_tools::lock_memory();
```
Comment (saikishor) [inline controller_manager/src/ros2_control_node.cpp:66]:
```suggestion
    const auto lock_result = realtime_tools::lock_memory();
```

**Review by saikishor — APPROVED** [review]
LGTM 

This looks like a duplicate of https://github.com/ros-controls/ros2_control/pull/1892

However, considering it as a first issue. It's fine

**Review by saikishor — APPROVED** [review]

**Review by christophfroehlich — APPROVED** [review]

**Review by christophfroehlich — CHANGES_REQUESTED** [review]
please fix the pre-commit issue (run `pre-commit run --all`)

**Comment by imsudiproy** [comment]
> please fix the pre-commit issue (run `pre-commit run --all`)

Fixed the pre-commit issue.
```
root@1bd3e3af4a7d:~/ros2_humble_ws/src/ros2_control# pre-commit run --all
check for added large files..............................................Passed
check python ast.........................................................Passed
check for case conflicts.................................................Passed
check docstring is first.................................................Passed
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
check xml................................................................Passed
check yaml...............................................................Passed
debug statements (python)................................................Passed
fix end of files.........................................................Passed
mixed line ending........................................................Passed
trim trailing whitespace.................................................Passed
fix utf-8 byte order marker..............................................Passed
pyupgrade................................................................Passed
pydocstyle...............................................................Passed
black....................................................................Passed
flake8...................................................................Passed
clang-format.............................................................Passed
ament_cppcheck...........................................................Passed
ament_cpplint............................................................Passed
ament_lint_cmake.........................................................Passed
ament_copyright..........................................................Passed
doc8.....................................................................Passed
rst ``code`` is two backticks............................................Passed
rst directives end with two colons.......................................Passed
rst ``inline code`` next to normal text..................................Passed
codespell................................................................Passed
Validate GitHub Workflows................................................Passed
Validate GitHub Actions..............................(no files to check)Skipped
Validate Dependabot Config (v2)..........................................Passed
```

**Review by bmagyar — APPROVED** [review]
thank you!

**Pull request merged**